src/python_framework_v02/troute/network/reservoirs/levelpool/levelpool.pyx

```diff
@@ -2,32 +2,44 @@ cimport numpy as np
 import numpy as np
 from libc.stdlib cimport malloc, free
 
+from troute.network.reach cimport compute_type, _Reach
 """
 Externally defined symbols
 """
-############ Level Pool Reservoir Interface ############
-cdef extern void* get_lp_handle() nogil;
 
-cdef extern void init_lp(void* handle, float *water_elevation, float *lake_area, float *weir_elevation,
-                    float *weir_coefficient, float *weir_length, float *dam_length, float *orifice_elevation,
-                    float *orifice_coefficient, float *orifice_area, float *max_depth, int *lake_number) nogil;
+cdef extern from "levelpool_structs.c":
+  void init_levelpool_reach(_Reach* reach, int lake_number,
+                            float dam_length, float area, float max_depth,
+                            float orifice_area, float orifice_coefficient, float orifice_elevation,
+                            float weir_coefficient, float weir_elevation, float weir_length,
+                            float initial_fractional_depth, float water_elevation
+  )
+  void free_levelpool_reach(_Reach* reach)
+  void route(_Reach* reach, float routing_period, float inflow, float lateral_inflow, float* outflow,  float* water_elevation) nogil
```

Comment:
should pick  a consistent interface name (run, route, ect) and use it across the various types (in fact, I'm not curious if cython types can be `ABC` (abstract base class) which could be useful to implement the `Reach` type as an `ABC` and mark the routing function as an `@abstract_method` so that all subclasses implement it.